src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -720,25 +720,10 @@ public void InsertDefault()
         }
 
         [Fact]
-        public void InsertRangeNoOpIdentity()
-        {
-            Assert.Equal(s_empty, s_empty.InsertRange(0, s_empty));
-            Assert.Equal(s_oneElement, s_empty.InsertRange(0, s_oneElement)); // struct overload
-            Assert.Equal(s_oneElement, s_empty.InsertRange(0, (IEnumerable<int>)s_oneElement)); // enumerable overload
-            Assert.Equal(s_oneElement, s_oneElement.InsertRange(0, s_empty));
-        }
-
-        [Fact]
-        public void InsertRangeEmpty()
+        public void InsertRangeEmptyInvalid()
         {
             Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(-1, 10));
             Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(1, 10));
-            Assert.Equal(new int[0], s_empty.InsertRange(0, Enumerable.Empty<int>()));
-            Assert.Equal(s_empty, s_empty.InsertRange(0, Enumerable.Empty<int>()));
-            Assert.Equal(new[] { 1 }, s_empty.InsertRange(0, new[] { 1 }));
-            Assert.Equal(new[] { 2, 3, 4 }, s_empty.InsertRange(0, new[] { 2, 3, 4 }));
-            Assert.Equal(new[] { 2, 3, 4 }, s_empty.InsertRange(0, Enumerable.Range(2, 3)));
-            Assert.Equal(s_manyElements, s_manyElements.InsertRange(0, Enumerable.Empty<int>()));
```

Comment:
@AArnott, in order to be fed to the `[Theory]` this had to be moved into its own member data at https://github.com/dotnet/corefx/pull/13158/files#diff-b5f51582b094b16996afaa39afabf9ebR796. [Same](https://github.com/dotnet/corefx/pull/13158/files#diff-b5f51582b094b16996afaa39afabf9ebR806) for the `InsertRangeNoOpIdentity` fact that was removed above.

If you want, I can simply keep these test cases while still adding the refactored versions down below, and then submit a PR that removes these since they're now redundant. However IMO, it seems fairly obvious to me that they're being covered by the new MemberDatas.